packages/pwa-devdocs/src/peregrine/reference/rest-api-client/index.md

```diff
@@ -0,0 +1,225 @@
+        parseJSON: false
+    }
+).then(({ response }) => console.log(response);)
+```
+
+### Error handling
+
+Promoises returned by `request()` are rejected if the server responds with an HTTP error code within the 4xx-5xx range.
+This is different from the standard `fetch()` behavior, which ignores HTTP errors so it can resolve the promise and only fails on network connectivity errors.
+
+The Magento REST API uses HTTP error codes to denote failed operations, so
+the `request()` method responds to these server-side errors by rejecting them.
+When the server returns an error code, the `request()` method builds an error containing metadata for debugging and rejects the `Promise`.
+
+
+### Multicasting
+
+The multicasting feature of the Magento 2 REST client collects all successive calls to the `request()` method and returns a single request and response that is shared to all method callers.
+
+In any distributed architecture, where applications are built from components with view, behavior, and data fetching strategy, multiple components may request the same resource at the same time. 
+GraphQL clients resolve this by merging GraphQL queries together, dispatching a request as a single call, and distributing the pieces of the GraphQL response to the original callers.
+A direct call using `fetch()` or `XMLHttpRequest()` does not provide this functionality.
+
+The `request()` method makes up for this by using a pattern-matching rule on inflight requests.
+All requests with the same method, path, and body text are considered multicastable by default.
+A clone of the response object is returned if a request meets the following criteria:
+
+* The pattern of the new request matches one that is already in progress, i.e. the network has not loaded the resource yet.
+* No unsafe operations, such as `POST`, have occurred to the target resource between the time of the original request and the new request.
+* The request is not a `POST` request.
+  These requests create and/or modify server state, so it is unsafe to reuse their results.
+
+#### Override multicast
+
+The `multicast` configuration option forces multicasting to be `true` or `false` and bypasses the previously described rules.
+
+```js
+// A normally multicast request forced not to multicast
+request('carts/mine',{ method: 'GET', multicast: false });
+
+// A normally non-multicast request forced to multicast
+request(
+    'carts/mine/items',
+    {
+        method: 'POST',
+        body: cartItem,
+        multicast: true
+    }
+);
+```
+
+### Rolling requests
+
+A rolling request is a multicast which overrides and replaces matching, previous multicast requests.
+It helps maintain a fresh resource, such as a shopping cart, and reduces network traffic by aborting outstanding matching requests.
+
+This feature has no effect on non-multicast requests.
+
+To use a rolling request, set the `cache` configuration option to `reload` or `no-store`.
+
+### The `Magento2ApiRequest` class
```

Comment:
This inconsistency is from me, not you, but in retrospect, we're going back and forth with calling this class `Magento2ApiRequest` and `M2ApiRequest`. In the code, the property on the `request` function is named `M2ApiRequest`, so we should replace all instances of `Magento2ApiRequest` with `M2ApiRequest`.